src/App.tsx

```diff
@@ -1,10 +1,72 @@
-/* eslint-disable jsx-a11y/label-has-associated-control */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
+import cn from 'classnames';
 import { UserWarning } from './UserWarning';
-import { USER_ID } from './api/todos';
+import { getTodos, USER_ID } from './api/todos';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+
+interface Errors {
+  messages: string[];
+}
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>();
+  const [errors, setErrors] = useState<Errors>({ messages: [] });
+  const [input, setInput] = useState('');
+  const [isLoadingTodos, setIsLoadingTodos] = useState(true);
+  const [isNotificationVisible, setNotificationVisible] = useState(false);
+  const [errorTimeout, setErrorTimeout] = useState<NodeJS.Timeout | null>(null);
+
+  const showError = useCallback(
+    (message: string) => {
+      setErrors(prev => ({
+        messages: [...new Set([...prev.messages, message])],
+      }));
+      setNotificationVisible(true);
+
+      if (errorTimeout) {
+        clearTimeout(errorTimeout);
+      }
+
+      const timeout = setTimeout(() => {
+        setErrors({ messages: [] });
+        setNotificationVisible(false);
+      }, 3000);
+
+      setErrorTimeout(timeout);
+    },
+    [errorTimeout],
+  );
+
+  const handleFormSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (input.trim().length === 0) {
+      showError('Title should not be empty');
+
+      return;
+    }
+
+    setInput('');
+    setErrors({ messages: [] });
+  };
+
+  useEffect(() => {
+    const fetchTodos = async () => {
+      try {
+        const todosData = await getTodos();
+
+        setTodos(todosData);
+      } catch (error) {
+        showError('Unable to load todos');
+      } finally {
+        setIsLoadingTodos(false);
+      }
+    };
+
+    fetchTodos();
```

Comment:
```suggestion
    (async () => {
      try {
        const todosData = await getTodos();

        setTodos(todosData);
      } catch (error) {
        showError('Unable to load todos');
      } finally {
        setIsLoadingTodos(false);
      }
    })();
```

https://developer.mozilla.org/en-US/docs/Glossary/IIFE